File: Backend/src/main/java/com/futurum/task/models/Product.java
package com.futurum.task.models;

import lombok.*;

import javax.persistence.*;
import java.math.BigDecimal;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@ToString
@Entity
@Table(name = "Products")
public class Product {
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Id
    private Long productID;
    private String productName;
    private BigDecimal productPrice;

    @OneToOne
    @JoinColumn(name = "campaignID")
    Campaign campaign;

    public Product(Long productID, String productName, BigDecimal productPrice) {
        this.productID = productID;
        this.productName = productName;
        this.productPrice = productPrice;
    }
}

File: Frontend/futurumTask/src/app/classes/product.ts
import { Campaign } from "./campaign";

export class Product {
    productID: number;
    productName: string;
    productPrice: number;
    campaign?: Campaign;
    
    constructor(productID : number, productName: string,
        productPrice: number, campaign? : Campaign){
        this.productID = productID;
        this.productName = productName;
        this.productPrice = productPrice;
        this.campaign = campaign;
    }

}
